plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/N4JSProjectSetupJsonValidatorExtension.xtend

```diff
@@ -683,33 +686,33 @@ public class N4JSProjectSetupJsonValidatorExtension extends AbstractJSONValidato
 
 		// collect all declared module filter specifiers
 		val nameValuePairs = collectObjectValues(moduleFiltersValue as JSONObject);
-		val filterSpecifierPairs = nameValuePairs.values
+		val filterSpecifierTraceables = nameValuePairs.values
 			.filter(JSONArray)
 			.flatMap[elements]
 			.filterNull
-			.map[ProjectDescriptionUtils.getModuleFilterSpecifier(it)->it]
+			.map[filter | ASTTraceable.of(filter, ProjectDescriptionUtils.getModuleFilterSpecifier(filter))];
 
-		holdsValidModuleSpecifiers(filterSpecifierPairs, project);
+		holdsValidModuleSpecifiers(filterSpecifierTraceables, project);
 	}
 	
 	
-	private def holdsValidModuleSpecifiers(Iterable<Pair<ModuleFilterSpecifier, JSONValue>> moduleFilterSpecifiers, IN4JSProject project) {
-		val validFilterSpecifier = new ArrayList<Pair<ModuleFilterSpecifier, JSONValue>>();
+	private def holdsValidModuleSpecifiers(Iterable<ASTTraceable<ModuleFilterSpecifier>> moduleFilterSpecifiers, IN4JSProject project) {
```

Comment:
How about moving the `ASTTraceable` to some general utils package in a general bundle or n4js bundle?